### ISSUE TITLE
Module import path error on Linux in production build.

### ISSUE THREAD

**Issue by blaher** [ember-cli/ember-cli#5638]
My issue seems pretty similar to issue [#5542](https://github.com/ember-cli/ember-cli/issues/5542), but I'm running in to the same issue on a linux instance. Tried building it through Jenkins on Amazon Linux AMI release 2015.09, got the error and then tried again manually with my own machine with Ubuntu 14.04.3 LTS and got the same thing.

I originally upgraded from 1.13.13 to 2.0.0 when it started happening. Since then upgraded to 2.4.3 to see if it would fix it. Just like the other issue, everything runs fine and builds correctly in development, it only happens in a production build. The error I get is the same:

`Error: Could not find module -private/system/references/record imported from ember-data/-private/system/references`
`...n(e){this.name=e}function i(e,t){throw new Error("Could not find module `"+e+"` ...`
(pointing to the "new")

Ember: 2.4.2
Ember Data: 2.4.1+4260f5771d
Ember-CLI: 2.4.2
Node: 5.6.0
NPM: 3.6.0

[Development Build](http://quality.maintenanceprogram.com/office/)
[Production Build](http://new.maintenanceprogram.com/office/)

Any way to help track what's going on?

**Comment by awaer** [comment]
Confirmed – I'm also seeing the same error when building for production. No errors when running `ember serve`. 

Could there be some build dependency that just updated and is now breaking builds?

**Comment by abercrave** [comment]
+1

Only `ember s` and `ember b` build a working site.

```
Ember: 2.4.2
Ember Data: 2.4.1+4260f5771d
jQuery: 2.2.2
Node: v5.5.0
NPM: 3.7.1
```

**Comment by chadhietala** [comment]
I believe this is actually an issue with Ember Data and was resolved in https://github.com/emberjs/data/pull/4214.

**Comment by blaher** [comment]
Was this resolved in Ember Data 2.4.3+a3e2521f85? I'm still having this problem.
